### DISCUSSION TITLE
Installation / versions / USBasp

### DISCUSSION THREAD

**Discussion by kratochviljan** [SpenceKonde/ATTinyCore#528] [Support / Q & A]
1.5.0 - Error downloading https://azduino.com/bin/micronucleus/micronucleus-cli-2.5-azd1-i686-mingw32.tar.bz2
Nothing was installed. -> https://i.imgur.com/0cyzoM7.png

1.4.1 - CRC doesn't match. File is corrupted.
Nothing was installed. -> https://i.imgur.com/5iAIIna.png

1.4.0 - CRC doesn't match. File is corrupted.
Nothing was installed. -> https://i.imgur.com/1ylVJIu.png

1.3.3 - CRC doesn't match. File is corrupted.
Nothing was installed. -> https://i.imgur.com/UZBCgHO.png

1.3.2 - Installation completed.
This is the latest version that can be installed (at least for me)  -> https://i.imgur.com/M6occS6.png
But there is another issue with USBasp programmer, generic option selecion for "USBasp" is gone, only "USBasp (ATTinyCore)" can be selected which unfortunately does not work-> 
_The uploader returned an error
avrdude: Warning: cannot query manufacturer for device: No such file or directory
avrdude: Warning: cannot query product for device: No such file or directory
avrdude: error: could not find USB device with vid=0x16c0 pid=0x5dc vendor='www.fischl.de' product='USBasp'_

frustrating, sorry...

**Comment by kratochviljan** [comment]
uaaaaaa -> error found!!!

It completely break AVR/GCC/AVRDude chain.
If you are using Arduino 1.6.9 for obvious reasons (like many many others) DO NOT install this over JSON / Board Manager it will completely break everything. I had to completely restore whole directory c:\users\xxx\AppData\Local\Arduino15 from backup to make everything work as before installation.

Frustrating? No -> ANGER!!!

**Reply by SpenceKonde** [reply]
Well, I think I see what must have happened here. Handling of toolchain versions specified by third party hardware packages appears to be totally broken in 1.6.9.... This is a defect in the IDE, not my core, and they've long since fixed it. Among other things, if I install a core which downloads a non-arduino-packaged version of avr-gcc (like my DxCore), that version will be used for everything  (not sure whether this part is unintended - **though it SHOULD BE**), if there is another core installed which pulled in any version of the Arduino-packaged avr-gcc toolchain, it will ignore what DxCore specifies in tools-dependencies and use the arduino version, despite it being years older and not supporting the DxCore parts. The IDE's handling here looks completely broken. Now that I think about it. back in those days, I remember having multiple portable installations for board manager installs , and that board manager installation of one package could break another one. That issue is fixed on recent versions of the IDE.

I advise installing 1.8.13 - or rather, using portable .zip-based version - I never run the installer, always .zip package (and I have like half a dozen versions on my main computer - which is down from I think a peak of 10 or so during the bad old days of 1.6.4~1.8.0, when every version broke at least one thing and fixed at least one thing. And back then, yeah, I think 1.6.9 was one of the versions that was preferable to the next several releases. Ever since late in the 1.6.x releases, though, it hasn't been like that.

**Comment by SpenceKonde** [comment]
Thanks for reporting this, I will investigate immediately, I was unaware
that there were any issues with any versions!!!

____________
Spence Konde
Azzy’S Electronics

New products! Check them out at tindie.com/stores/DrAzzy
GitHub: github.com/SpenceKonde
ATTinyCore: Arduino support for almost every ATTiny microcontroller
Contact: spencekonde@gmail.com

**Comment by SpenceKonde** [comment]
Can you get the text of those error messages about SSL showing in the
background? Thanks

____________
Spence Konde
Azzy’S Electronics

New products! Check them out at tindie.com/stores/DrAzzy
GitHub: github.com/SpenceKonde
ATTinyCore: Arduino support for almost every ATTiny microcontroller
Contact: spencekonde@gmail.com

**Comment by SpenceKonde** [comment]
What the hell.... so this is blowing up old ide versions? I thought I had
the CI scripts running with 1.6.9 too

____________
Spence Konde
Azzy’S Electronics

New products! Check them out at tindie.com/stores/DrAzzy
GitHub: github.com/SpenceKonde
ATTinyCore: Arduino support for almost every ATTiny microcontroller
Contact: spencekonde@gmail.com

**Comment by SpenceKonde** [comment]
The USBAsp issue is endemic to recent versions of AVRdude, which are needed to fix other problems if you have the wrong version of the drivers installed, the well established fix is to use Zadig to swap to the correct USBAsp driver. 

1. Download Zadig from http://zadig.akeo.ie
2. Plug in USBasp
3. Start zadig
4. Options > List all devices
5. Select USBasp from the drop down menu
6. Select libusbK(v3.0.7.0) driver
7. Click Install

**Comment by SpenceKonde** [comment]
I've created issues #529 and #530 in response to this - if you are able to get me the text of those errors that showed up in the console, that would be very helpful. I can't actually work on this until done with day job today. I would also very much like more information on what you mean about it breaking the the toolchain. I am certainly very distressed to hear that four versions of the core that I belived were working are not even installing - and even moreso that it is claimed to be blowing up a version of the IDE which I believed it was compatible with! 

And for that matter, what the "obvious reasons" are for using 1.6.9?  I thought the IDE had gotten through the rough patch when half of the releases sucked for one reason or another, and have been recommending 1.8.13, and before that, "latest 1.8.x" for a while. They really got their act together in the 1.8.x versions.

**Reply by sleemanj** [reply]
> I thought the IDE had gotten through the rough patch when half of the releases sucked for one reason or another

Indeed.  My memory is hazy now, but 1.6.13 was about where things started to get better.  This "stuff got nuked" problem was happening up to 1.6.10 I think.

On my fork's distribution I have the following warning.

---

The current release was made using IDE version 1.8.5 so that's the recommended, but you should be ok with anything 1.6.13 or newer, and MAYBE older versions.

:boom: ** Do not use Arduino IDE 1.6.10 due to major bugs in that version of the IDE. :boom:  

---

**Comment by SpenceKonde** [comment]
Both issues are closed. #529 is due to bugs in 1.6.9 of the IDE; the solution is to use a newer IDE. 
Issue #530 was due to the improper isolation between cores with different versions of their dependencies. This is an IDE bug which is fixed in modern versions of the IDE. 

This is why we recommend 1.8.13 rather than old buggy versions like 1.6.9. 1.6.9 *was* the best version for the next several releases, IIRC, but it has long since been surpassed by the actually-working 1.8.x versions.
